### PR TITLE
Adds a new illegal-tech handcuff circuit shell.

### PR DESCRIPTION
**Pull request by Wallemations**
## About The Pull Request
Adds a handcuff circuit shell, which outputs the entity it is currently restraining, the amount of time it is set to be removed in, and a trigger both when it is added and removed from a person.
To print it requires that illegal tech has been researched, since I doubt giving literally everyone handcuffs that act as triggers would be a good idea.

Here's the sprite
We have become the death stranding sam

## Why It's Good For The Game
Lets crazy people make more elaborate jigsaw traps.
I guess it also can be used to instantly have sec be told if an officer is arresting someone but jigsaw traps are funnier.

## Changelog

:cl: Wallem
add: The blueprints for a banned handcuff circuit shell has been covertly slipped into the station's illegal tech tree.
/:cl:

### CODE REVIEW

**Comment by TheBonded** [comment]
This is illegal tech so itll never get made. Maybe put it under a separate new tech after advanced circuitry and basic security tech, and have the shell only printable at security techfabs? Besides, everyone has access to cable coil anyways if they want to make handcuffs.

**Comment by Wallemations** [comment]
> This is illegal tech so itll never get made. Maybe put it under a separate new tech after advanced circuitry and basic security tech, and have the shell only printable at security techfabs? Besides, everyone has access to cable coil anyways if they want to make handcuffs.

An entirely new tech just for a circuit shell seems a bit weird to me, though I get where you're coming from.

File: code/modules/research/techweb/all_nodes.dm [hunk 1]
```diff
@@ -1981,6 +1981,7 @@
 		"largecrossbow",
 		"rapidsyringe",
 		"suppressor",
+		"handcuff_shell",
```
Comment (Mothblocks) [inline code/modules/research/techweb/all_nodes.dm:1984]: ABC order

**Comment by Mothblocks** [comment]
@Watermelon914 We're going to have to discuss this (over Discord or something) and whether or not it breaks non-goal 3 or not

File: code/modules/research/techweb/all_nodes.dm (hunk 1 above)
Comment (Wallemations) [reply code/modules/research/techweb/all_nodes.dm:1984]: Ah, forgot. fixin that then

**Comment by Wallemations** [comment]
> whether or not it breaks non-goal 3 or not

Shit I forgot about that. 
Maybe making it that sec-protolate-only thing would actually make it better for that?
In that case it sec could use it to make a loudspeaker that announces arrests in the brig or something cool like that.

**Comment by Wallemations** [comment]
Having thought more about it I've really turned to like looking at these as a sec tool more than a tool for evil people.
Sec could set up jump pads to instantly teleport perps to the brig, announce arrests, n' maybe some other cool shit I can't even think of idfk.

I know you guys are gonna talk about this PR so if I get the go-ahead I was wondering if it would be a good change to make it a sec protolathe thing or if it would be best as an illegal tech.
Asking for your guys' input since I don't wanna break any rules/goals.

**Comment by Mothblocks** [comment]
If it breaks one of the non-goals, then it would need to be closed in order to keep circuit power in check.

That being said, I genuinely don't know, this seems like something that's okay? Except they'd need to be only ever as good as normal handcuffs, meaning their breakout time can't be any longer. Will talk to Watermelon about it.

**Comment by Wallemations** [comment]
> If it breaks one of the non-goals, then it would need to be closed in order to keep circuit power in check.
> 
> That being said, I genuinely don't know, this seems like something that's okay? Except they'd need to be only ever as good as normal handcuffs, meaning their breakout time can't be any longer. Will talk to Watermelon about it.

Their breakout time has a clamp at 60 seconds, which is normal time. If you want I could bring it down to ziptie time, like 45 seconds.

**Comment by moocowswag** [comment]
The only unique use I could actually think of so far is that its a circuit that you attach to someone with only a short do-after.

So with just a short shove stun -> cuff -> grab grab you have successfully cuffed someone .... and now have perfect location tracking of them until they take the cuffs off




That being said they might be useful for a gimmick, like for a jigsaw trap or forced hunger games if you capture multiple people and I would probably tc trade for them rather than trying to helplessly attempt to get illegal tech.

**Comment by Mothblocks** [comment]
> That being said they might be useful for a gimmick, like for a jigsaw trap or forced hunger games if you capture multiple people 

Ya this would be the hope

**Comment by moocowswag** [comment]
Oh and I`m not sure if this will work but maybe putting plasma in the cell of the circuit and then sending it a signal so it blows up? So you could cuff people and have them answer questions and if they get it wrong they blow up? i dont know.

**Comment by Wallemations** [comment]
> > That being said they might be useful for a gimmick, like for a jigsaw trap or forced hunger games if you capture multiple people 
> 
> 
> 
> Ya this would be the hope

I'll admit that's the idea I had when I made them, but idk if that counts as "a weapon" or not.

**Comment by Wallemations** [comment]
> I would probably tc trade for them rather than trying to helplessly attempt to get illegal tech. 

To be (additionally) honest I did illegal tech because I thought emagging a research console unlocked illegal tech, turns out it just makes it so you can fuck with the R&D Server

**Comment by Mothblocks** [comment]
Off hand, can you make this require a real pair of handcuffs? That way you're only ever nerfing an existing pair.

**Comment by Mothblocks** [comment]
Like, what you print it out is a half-finished shell that you finish with a real pair of handcuffs

**Comment by moocowswag** [comment]
Watch this somehow end up letting you upgrade /obj/item/restraints/handcuffs/fake into real handcuffs when you add a circuit to them.

**Comment by Mothblocks** [comment]
Lol, good catch--yeah make sure it checks specifically

**Comment by Wallemations** [comment]
> Like, what you print it out is a half-finished shell that you finish with a real pair of handcuffs

Good idea, I'll do that RQ

**Comment by ATH1909** [comment]
If the lore is that the Kheiral cuffs are meant to be used as wrist communicators but are often just used as handcuffs, would it be possible to make the cuffs able to be put on someone without restraining them if you haven't inserted handcuffs into them? It seems a bit odd that we can't actually use the Kheiral cuffs for their "intended" purpose.

**Comment by Wallemations** [comment]
> If the lore is that the Kheiral cuffs are meant to be used as wrist communicators but are often just used as handcuffs, would it be possible to make the cuffs able to be put on someone without restraining them if you haven't inserted handcuffs into them? It seems a bit odd that we can't actually use the Kheiral cuffs for their "intended" purpose.

I initially had thought of something like that, just making them able to be worn in the glove slot or something as a bracelet like the ones in DS. It'd be a bit confusing to figure out balance-wise though, if it's a communication device. Maybe I could just make them turn into a GPS when in your gloves slot or something, which would make sense given what they're referencing in the first place.

**Comment by Wallemations** [comment]
> Maybe I could just make them turn into a GPS when in your gloves slot or something, which would make sense given what they're referencing in the first place.

Wait a fucking second that's actually a really good idea, it could be a mining tool then.
This could be a mining vendor thing that needs points!
You'd still need to get handcuffs from sec to finish them, so that's relatively balanced.

That seems like a _**much**_ better way to get them than illegal tech

**Pull request closed without merging**